membership_variable_period/models/product_template.py

```diff
@@ -79,3 +88,15 @@ def write(self, vals):
             )
             super(ProductTemplate, rec).write(vals2)
         return True
+
+    @api.constrains(
+        "membership_date_from",
+        "membership_date_to",
+        "membership",
+        "membership_type",
+    )
+    def _check_membership_dates(self):
+        for record in self:
+            if record.membership:
+                if record.membership_type == "fixed":
+                    return super(ProductTemplate, record)._check_membership_dates()
```

Comment:
You're breaking your loop here. Maybe what you want to do is:

```suggestion
        return super(ProductTemplate, self.filtered(lambda x: x.membership_type == "fixed"))._check_membership_dates()
```

Reply:
Good catch!